clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -469,17 +495,18 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                     TESTANY(OFFLINE_FILE_TYPE_FILTERED | OFFLINE_FILE_TYPE_IFILTERED,
                             shard->file_type_),
                 "Signal handler return point incorrect");
-            // We assume paired signal entry-exit (so no longjmp and no rseq
-            // inside signal handlers).
-            shard->prev_xfer_int_pc_.pop();
-            shard->pre_signal_instr_.pop();
         }
+        // It is a little inefficient to replace the top instr on the stack everytime.
+        // But we cannot perform this book-keeping using prev_instr_ on a
+        // TRACE_MARKER_TYPE_KERNEL_EVENT marker. If there was no instruction between
+        // two nested signals, we do not want to record any pre-signal instr for the
+        // second signal.
```

Comment:
"between two nested signals": there is no test of this added here so I wonder if you mean "between two consecutive signals at the same scope" (i.e., not nested wrt each other).

Reply:
We have test of both cases: see "Nested signals without any intervening instr." and "Multiple nested signals without any intervening instr between the nested".

> I wonder if you mean "between two consecutive signals at the same scope" (i.e., not nested wrt each other).

That's correct. I'll rephrase it to be clearer.

Reply:
Actually this comment applies to both: nested signals and consecutive signals. I clarified in the comment.